ArduCopter/mode_autorotate.cpp

```diff
@@ -134,25 +199,29 @@ void ModeAutorotate::run()
 
             }
 
-            // Slowly change the target head speed until the target head speed matches the parameter defined value
-            if (g2.arot.get_rpm() > HEAD_SPEED_TARGET_RATIO*1.005f  ||  g2.arot.get_rpm() < HEAD_SPEED_TARGET_RATIO*0.995f) {
+            if(!hover_autorotation){ 
+			// Slowly change the target head speed until the target head speed matches the parameter defined value
+           if (g2.arot.get_rpm() > HEAD_SPEED_TARGET_RATIO*1.005f  ||  g2.arot.get_rpm() < HEAD_SPEED_TARGET_RATIO*0.995f) {
                 _target_head_speed -= _hs_decay*G_Dt;
             } else {
                 _target_head_speed = HEAD_SPEED_TARGET_RATIO;
             }
-
-            // Set target head speed in head speed controller
-            g2.arot.set_target_head_speed(_target_head_speed);
-
-            // Run airspeed/attitude controller
-            g2.arot.set_dt(G_Dt);
-            g2.arot.update_forward_speed_controller();
-
-            // Retrieve pitch target
-            _pitch_target = g2.arot.get_pitch();
-
-            // Update controllers
-            _flags.bad_rpm = g2.arot.update_hs_glide_controller(G_Dt); //run head speed/ collective controller
+			  g2.arot.get_collective_minimum_drag(motors->get_coll_mid());
+               // Set target head speed in head speed controller
+               g2.arot.set_target_head_speed(_target_head_speed);
+               // Run airspeed/attitude controller
+               g2.arot.set_dt(G_Dt);
+               g2.arot.update_forward_speed_controller();
+               // Retrieve pitch target
+               _pitch_target = g2.arot.get_pitch();
+               // Update controllers
+              _flags.bad_rpm = g2.arot.update_hs_glide_controller(G_Dt); //run head speed/ collective controller
+                }else{
+            	_pitch_target = 0.0f;
+            	g2.arot.set_collective_minimum_drag(motors->get_coll_mid());
```

Comment:
This sets the collective directly into the motors class which modifies the filter freq too.   If we plan to use this more than once, why not call for it in the init where it might be used the first time.  Then it is available and can be set in the event of a hover auto.

Reply:
separately, why not initiate the touchdown controller since this portion of code will only be seen for one cycle.  I doubt this collective setting barely has time to take effect.  Just call the touchdown controller and set the collective before hand so as to have it start at the minimum drag collective.

Reply:
Replaced with a new function that handles collective for hovering autorotation.